src/snakemake/logging.py

```
@@ -656,38 +660,36 @@
-        if self.mode == ExecMode.DEFAULT and not self.settings.dryrun and logfiles:
+        if self.settings.enable_file_logging and not self.settings.dryrun and logfiles:
             log_paths = ", ".join([os.path.abspath(p) for p in logfiles])
             self.logger.info(f"Complete log(s): {log_paths}")
         return logfiles
 
     def cleanup_logfile(self):
-        from snakemake_interface_executor_plugins.settings import ExecMode
-
-        if self.mode == ExecMode.DEFAULT:
+        if self.settings.enable_file_logging:
             for handler in self.logfile_handlers.keys():
```

Ruff (0.12.2)

Use `key in dict` instead of `key in dict.keys()`

Remove `.keys()`

(SIM118)